### PR TITLE
Add  SQL Dine DNS Policy

### PR DESCRIPTION
**Pull request by marvinbuss**
## Summary of the Pull Request
* Add  SQL Dine DNS Policy

## PR Checklist
* [x] Closes Issue #xxx
* [x] Code of Conduct signed. If not, go to [Code of Conduct](https://github.com/microsoft/industry/blob/main/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-CorpLzPolicySetDefinition.json [hunk 1]
```diff
@@ -10,7 +10,133 @@
...
+                                "details": {
+                                    "type": "Microsoft.Network/privateEndpoints/privateDnsZoneGroups",
+                                    "roleDefinitionIds": [
+                                        "/providers/Microsoft.Authorization/roleDefinitions/4d97b98b-1d4f-4787-a291-c67834d212e7"
+                                    ],
+                                    "existenceCondition": {
+                                        "allOf": [
+                                            {
+                                                "field": "Microsoft.Network/privateEndpoints/privateDnsZoneGroups/privateDnsZoneConfigs[*].privateDnsZoneId",
+                                                "equals": "[parameters('privateDnsZoneId')]"
```
Comment (krnese) [inline foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-CorpLzPolicySetDefinition.json:80]: Please escape the ARM template functions properly using this pattern "[[parameter('paramName')]" if not, it is not deployable.

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-CorpLzPolicySetDefinition.json [hunk 2]
```diff
@@ -10,7 +10,133 @@
...
+                                                    "location": {
+                                                        "type": "string"
+                                                    }
+                                                },
+                                                "resources": [
+                                                    {
+                                                        "name": "[concat(parameters('privateEndpointName'), '/deployedByPolicy')]",
+                                                        "type": "Microsoft.Network/privateEndpoints/privateDnsZoneGroups",
+                                                        "apiVersion": "2020-03-01",
+                                                        "location": "[parameters('location')]",
```
Comment (krnese) [inline foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-CorpLzPolicySetDefinition.json:106]: Please escape the ARM template functions properly using this pattern "[[parameter('paramName')]" if not, it is not deployable.

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-CorpLzPolicySetDefinition.json [hunk 3]
```diff
@@ -10,7 +10,133 @@
...
+                                                    "privateEndpointName": {
+                                                        "type": "string"
+                                                    },
+                                                    "location": {
+                                                        "type": "string"
+                                                    }
+                                                },
+                                                "resources": [
+                                                    {
+                                                        "name": "[concat(parameters('privateEndpointName'), '/deployedByPolicy')]",
```
Comment (krnese) [inline foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-CorpLzPolicySetDefinition.json:103]: Please escape the ARM template functions properly using this pattern "[[parameter('paramName')]" if not, it is not deployable.

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-CorpLzPolicySetDefinition.json [hunk 4]
```diff
@@ -10,7 +10,133 @@
...
+                                                        "type": "Microsoft.Network/privateEndpoints/privateDnsZoneGroups",
+                                                        "apiVersion": "2020-03-01",
+                                                        "location": "[parameters('location')]",
+                                                        "properties": {
+                                                            "privateDnsZoneConfigs": [
+                                                                {
+                                                                    "name": "privateDnsZone",
+                                                                    "properties": {
+                                                                        "privateDnsZoneId": "[parameters('privateDnsZoneId')]"
```
Comment (krnese) [inline foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-CorpLzPolicySetDefinition.json:112]: Please escape the ARM template functions properly using this pattern "[[parameter('paramName')]" if not, it is not deployable.

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-CorpLzPolicySetDefinition.json [hunk 5]
```diff
@@ -10,7 +10,133 @@
...
+                                                                    }
+                                                                }
+                                                            ]
+                                                        }
+                                                    }
+                                                ]
+                                            },
+                                            "parameters": {
+                                                "privateDnsZoneId": {
+                                                    "value": "[parameters('privateDnsZoneId')]"
```
Comment (krnese) [inline foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-CorpLzPolicySetDefinition.json:122]: Please escape the ARM template functions properly using this pattern "[[parameter('paramName')]" if not, it is not deployable.

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-CorpLzPolicySetDefinition.json [hunk 6]
```diff
@@ -10,7 +10,133 @@
...
+                                                },
+                                                "privateEndpointName": {
+                                                    "value": "[field('name')]"
```
Comment (krnese) [inline foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-CorpLzPolicySetDefinition.json:125]: Please escape the ARM template functions properly using this pattern "[[parameter('paramName')]" if not, it is not deployable.

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-CorpLzPolicySetDefinition.json [hunk 7]
```diff
@@ -10,7 +10,133 @@
...
+                                                },
+                                                "location": {
+                                                    "value": "[field('location')]"
```
Comment (krnese) [inline foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-CorpLzPolicySetDefinition.json:128]: Please escape the ARM template functions properly using this pattern "[[parameter('paramName')]" if not, it is not deployable.

**Comment by marvinbuss** [comment]
Sorry about that.
image

**Comment by marvinbuss** [comment]
Closing because of https://github.com/microsoft/industry/pull/384

**Pull request closed without merging**